### ISSUE TITLE
Error when reloading integration from HA

### ISSUE THREAD

**Issue by raulgbcr** [raulgbcr/lednetwf_ble#7]
Currently investigating this, @8none1 can you reproduce it? Just reload the integration on HA and check logs, leaves the integration in a broken state

(latest commit from main)

```
Logger: homeassistant.config_entries
Source: config_entries.py:565
First occurred: 3:25:02 PM (1 occurrences)
Last logged: 3:25:02 PM
Error unloading entry LEDnetWF02001D1F2023 (2023) for light

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 565, in async_unload
    await self._async_process_on_unload(hass)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 703, in _async_process_on_unload
    if job := self._on_unload.pop()():
              ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not callable
```

**Comment by raulgbcr** [comment]
v0.0.4 doesn't seem to have this problem, so its something related to the live updates changes

**Comment by 8none1** [comment]
Curious.  I don't know why that sleep would effect unloading.  I'll play around and see what's going on.

**Comment by 8none1** [comment]
I tried commenting out the `sleep` but I still see the error.  I think this is coming from somewhere inside the `config_flow`.  Still trying to understand it.  I wish there was a really good example of how to do all this stuff :)

**Comment by 8none1** [comment]
See if this works for you:  https://github.com/raulgbcr/lednetwf_ble/tree/8none1/reload_fix

It seems OK for me.

**Comment by raulgbcr** [comment]
Honestly, I would like to avoid adding all of that, feels like we are just cluttering this. 

Im testing with it, but the changes you made on the `update` function on `lednetwf.py` seems to be the problem for the error, I'm not entirely sure if its the delay or the `self._is_on = None` line, but rolling back to the way it was before the changes to this function seem to fix the problem entirely, so I would like to avoid adding more stuff if possible

**Comment by 8none1** [comment]
Not sure which "more stuff" you're referring to here.

The changes made to fix the problem are pretty straightforward:

https://github.com/raulgbcr/lednetwf_ble/compare/main...8none1/reload_fix#diff-6cb705fece238816ba60c2ca48e60f773a8e63450f08a8570dcc7fec4b080accR21

This is a formatting change to align more closely with what LED BLE does.  No functional change.

https://github.com/raulgbcr/lednetwf_ble/compare/main...8none1/reload_fix#diff-6cb705fece238816ba60c2ca48e60f773a8e63450f08a8570dcc7fec4b080accR43

This removes the device from the devices list when unloaded.  This prevents errors such as "device already configured" and is, AFAICT, more correct.

https://github.com/raulgbcr/lednetwf_ble/compare/main...8none1/reload_fix#diff-b9e73274b38f19bfed21dff4f83fcbba3df84fddb5b4e202eed16ae82ad82958R395
This just shortens the sleep

https://github.com/raulgbcr/lednetwf_ble/compare/main...8none1/reload_fix#diff-288c4f9ac575b1d3d28f67f912ab2edf6985dd05bee324d37b3a82461305ea51R40
This is the actual fix for the `None type is not callable` bug.  I think we were removing the device twice.

**Comment by 8none1** [comment]
In fact, with not tearing down the connection at set up, we can probably remove the sleep altogether now.  I'll test it.

**Comment by 8none1** [comment]
![image](https://github.com/raulgbcr/lednetwf_ble/assets/6552931/dcd1555b-96b8-4753-b3ee-3f2cfab8a920)

Seems like I already was testing it. :)

**Comment by raulgbcr** [comment]
> [main...8none1/reload_fix#diff-288c4f9ac575b1d3d28f67f912ab2edf6985dd05bee324d37b3a82461305ea51R40](https://github.com/raulgbcr/lednetwf_ble/compare/main...8none1/reload_fix#diff-288c4f9ac575b1d3d28f67f912ab2edf6985dd05bee324d37b3a82461305ea51R40) This is the actual fix for the `None type is not callable` bug. I think we were removing the device twice.

Thats what i thought, but rolling back that function specifically solves the problem, which doesn't make too much sense to me, probably its just me failing to understand whats going wrong with it. I just thought something was wrong with it, so I wanted to try and guess that first before checking the other unloads, my bad

> ![image](https://private-user-images.githubusercontent.com/6552931/287378565-dcd1555b-96b8-4753-b3ee-3f2cfab8a920.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0NjM4NTAsIm5iZiI6MTcwMTQ2MzU1MCwicGF0aCI6Ii82NTUyOTMxLzI4NzM3ODU2NS1kY2QxNTU1Yi05NmI4LTQ3NTMtYjNlZS0zZjJjZmFiOGE5MjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDFUMjA0NTUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDJlMDdiNDQ3NjUxMzYyMTQzZWJhNzliZjgwYjY2ZGViMzY1NmM0OWUyYzRjZmM4ZWFhZjE1NmE5MDliOTZiZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dV7WfRkUvucnuL-12oS_eQoUl-l3qwC4YAwHXNLGVgk)
> 
> Seems like I already was testing it. :)

I've tried that and can't get the live updates to work when its commented, not that I can get them to work continously, sometimes they just update sporadically, not sure if I messed something up with my installation 😅

**Comment by raulgbcr** [comment]
Just tried the branch too and seem to be working yep, thanks!!

Still I can't get my head on why reverting that function fixes it too, but anyway, feel free to merge it if you wish, thanks again.

**Comment by 8none1** [comment]
> not that I can get them to work continuously, sometimes they just update sporadically, not sure if I messed something up with my installation

I have a theory that these controllers are super flaky at BLE.  I have deployed ESPhome BLE relays to within a few meters of each of them, and sometimes they just stop being reachable.  I have lights based on the "Trinones" hardware that has a range of 10s of meters, these LEDnetWF ones seem to be more like a meter on a good day and 30cm on a bad day.

Still, we must persevere. :D

I will continue to play around with it today and make sure I haven't introduce any new bugs, but I actually feel pretty good about this now.  I've added a few comments to that future travellers can see what happened and why.
